### ISSUE TITLE
Input element loses focus after 1 character is typed (Authenticator re-rendering issue)

### ISSUE THREAD

**Issue by nickarocho** [aws-amplify/amplify-ui#1451]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

Create React App

#### Basic issue

When wrapping my app with the `<Authenticator>` component, the inputs lose focus after typing one character - forcing the user to refocus the input to continue typing.

#### Example

https://user-images.githubusercontent.com/16296496/156466157-1523431e-7f19-447b-bc4c-6f8b1ef01fa3.mov

#### Ideas/Workarounds
After doing a little digging, it sounds like the Authenticator component is forcing the blur on every re-render.

Here's a couple (kinda) similar StackOverflow questions related:
- https://stackoverflow.com/questions/59715158/react-hooks-input-loses-focus-when-1-character-is-typed-in
- https://stackoverflow.com/questions/42573017/in-react-es6-why-does-the-input-field-lose-focus-after-typing-a-character

Works as expected when using the `withAuthenticator` approach (via ["Quick Start" guide](https://ui.docs.amplify.aws/components/authenticator#quick-start)).

### What's the expected behaviour?

The user should be able to type fluidly into the input without having to manually refocus after every keystroke.

### Help us reproduce the bug!

1. Follow the tutorial building a full stack Amplify/React "todo" app at https://docs.amplify.aws/start/getting-started/setup/q/integration/react/
2. After completing the ["Create login UI" section](https://docs.amplify.aws/start/getting-started/auth/q/integration/react/#create-login-ui), try typing into the inputs (this is where the bug should be)
3. Swap out the `Authenticator` with `withAuthenticator` (like [this](https://ui.docs.amplify.aws/components/authenticator#quick-start))
4. Type into the inputs again - should work.

### Code Snippet

- Demo repo with bug: https://github.com/nickarocho/auth-playground/commit/31ce1660d556e66bab1a2a9ee82c74f17fb405bc
- Demo repo with workaround fix: https://github.com/nickarocho/auth-playground/commit/85a3bfa255a71213fe24ec8c0f346aee80ef9503

#### Doesn't Work

```js
import { Authenticator } from '@aws-amplify/ui-react';

function App() {
  // ...other logic hidden for brevity 
  return (
    <Authenticator>
      {({ signOut, user }) => (
        <div style={styles.container}>
          <div style={styles.row}>
            <button style={styles.button} onClick={signOut}>
              Sign out
            </button>
            <button style={styles.button} onClick={clear}>
              Clear
            </button>
          </div>
          <br />
          <h2>Amplify Todos</h2>
          <input
            onChange={(e) => setInput("name", e.target.value)}
            style={styles.input}
            value={formState.name}
            placeholder="Name"
          />
          <input
            onChange={(e) => setInput("description", e.target.value)}
            style={styles.input}
            value={formState.description}
            placeholder="Description"
          />
          <button style={styles.button} onClick={addTodo}>
            Create Todo
          </button>
          {todos.map((todo, index) => (
            <div key={todo.id ? todo.id : index} style={styles.todo}>
              <p style={styles.todoName}>{todo.name}</p>
              <p style={styles.todoDescription}>{todo.description}</p>
            </div>
          ))}
        </div>
      )}
    </Authenticator>
  );
}
```

#### Works:
```js
import { withAuthenticator } from '@aws-amplify/ui-react';

function App({ signOut, user }) {
  // ...other logic hidden for brevity 
  return (
    <div style={styles.container}>
      <div style={styles.row}>
        <button style={styles.button} onClick={signOut}>
          Sign out
        </button>
        <button style={styles.button} onClick={clear}>
          Clear
        </button>
      </div>
      <br />
      <h2>Amplify Todos</h2>
      <input
        onChange={(event) => setInput("name", event.target.value)}
        style={styles.input}
        value={formState.name}
        placeholder="Name"
      />
      <input
        onChange={(event) => setInput("description", event.target.value)}
        style={styles.input}
        value={formState.description}
        placeholder="Description"
      />
      <button style={styles.button} onClick={addTodo}>
        Create Todo
      </button>
      {todos.map((todo, index) => (
        <div key={todo.id ? todo.id : index} style={styles.todo}>
          <p style={styles.todoName}>{todo.name}</p>
          <p style={styles.todoDescription}>{todo.description}</p>
        </div>
      ))}
    </div>
  );
}

export default withAuthenticator(App)
```

### Additional information and screenshots

```json
// package.json
{
  "name": "auth-playground",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@aws-amplify/ui-react": "^2.7.0",
    "@testing-library/jest-dom": "^5.16.2",
    "@testing-library/react": "^12.1.3",
    "@testing-library/user-event": "^13.5.0",
    "@types/jest": "^27.4.1",
    "@types/node": "^17.0.21",
    "@types/react": "^17.0.39",
    "@types/react-dom": "^17.0.12",
    "aws-amplify": "^4.3.15",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-scripts": "5.0.0",
    "typescript": "^4.6.2",
    "web-vitals": "^2.1.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  }
}

```

**Comment by reesscot** [comment]
This was fixed in [2.9.0](https://www.npmjs.com/package/@aws-amplify/ui-react/v/2.9.0)

**Comment by nickarocho** [comment]
Wow that was fast, thanks Scott!
